Sources/AuthFoundation/OAuth2/OAuth2Client.swift

```diff
@@ -421,45 +421,51 @@ public final class OAuth2Client {
                 }
             }
         }
-        
-        // Exchange the token
-        request.send(to: self) { result in
-            guard case let .success(response) = result else {
-                completion(result)
-                return
-            }
-            
-            // Perform idToken/accessToken validation
-            do {
-                try response.result.validate(using: self, with: request as? IDTokenValidatorContext)
-            } catch {
-                completion(.failure(.validation(error: error)))
-                return
-            }
-            
-            // Wait for the JWKS keys, if necessary
-            group.notify(queue: DispatchQueue.global()) {
-                guard let idToken = response.result.idToken else {
-                    completion(result)
-                    return
-                }
-                
-                guard let keySet = keySet else {
-                    completion(.failure(.validation(error: JWTError.invalidKey)))
-                    return
-                }
+
+        openIdConfiguration { result in
```

Comment:
Good point. This had me thinking as well, on the off chance openId configuration isn't cached then that could be an indication that  something has gone wrong with the flow and we should throw an error. since the flow is triggered  at the `start` or at `resume` and they both call to `openIdConfiguration()`

Reply:
That's true, though we may change the model for how caching of openIdConfiguration works in the future. So it's better for us to protect against it, and ensure the operations that _could_ be async don't block the token exchange.

For more context, we fetch the JWKS keys in parallel with the token exchange; they don't require one-another, but we need both to complete in order to validate the returned tokens. As a result, it makes sense to use that same parallelization system to fetch configuration as well.